network_stresser_image/traffic_generator.py

```diff
@@ -85,13 +87,47 @@ def request_udp(self, amount, counter, silent):
         sock.sendto(data, (self.server, port))
         sock.close()
 
+    def request_iperf_tcp(self, amount, counter, silent):
+        """
+        Send an iPerf TCP request of the given amount to the server
+        @param amount: The amount of data to be sent
+        @param counter: The index of the current request in the test
+        @param silent: Should logging output be suppressed
+        """
+        dots, data = self.get_data(amount)
+        port = self.tcp_port
+        iperf = self.iperf
+        server = self.server
+        protocol = "IPERF TCP"
+        timestamp = now()
+        if not silent: print self.send_message.format(**locals()) + "\n"
+
+        system("{iperf} -c {server} -p {port}".format(**locals()))
+
+    def request_iperf_udp(self, amount, counter, silent):
```

Comment:
this is an entire code duplication - generalize the implementation above - accepting an extra parameter - is_udp